### ISSUE TITLE
[r] Use `libtiledbsoma` handles for arrays

### ISSUE THREAD

**Issue by johnkerl** [single-cell-data/TileDB-SOMA#3059]
Already done for groups

#2406; [[sc-55685]](https://app.shortcut.com/tiledb-inc/story/55685/r-use-libtiledbsoma-for-array-handles).

Steps:

* Status quo is
  * There is a `private$.tiledb_array` with connection to `self$object`
    * This is set on construction/open, and closed on close
  * The `libtiledbsoma` handles are currently ephemeral
    * Every `read` / `write` / `get_metadata` etc which is `libtiledbsoma`-capable currently takes a `uri` and a `ctx` and does open/op/close
* Needing to be done:
  * Remove the last refrences to `private$.tiledb_array` with connection to `self$object`
    * #3129
  * Glitch: while with 3129 we are removing the last references to `self$object` within `R/*.R`, there are hundreds of lines of uses of `$object` within `tests/testthat` -- tracked on #3124
  * Then set a `libtiledbsoma` handle on construction/open, and close it on close
  * Modify all the methods to not take `uri`+`ctx` but rather the open handle

To check:

* Currently the array `set_metadata` does  not error out when the array is opened for read -- this is a bug.
  * This should be auto-fixed by this issue, but, it needs to be tested.

**Comment by johnkerl** [comment]
Blocked by #3051 since here
https://github.com/single-cell-data/TileDB-SOMA/blob/1.15.0rc2/apis/r/R/SOMADataFrame.R#L182
the `self$object` is used. Removing that `private$.tiledb_array` / `self$object` from the base class `TileDBArray.R` causes this query-condition logic to return empty results, in a way that makes unit-test cases fail in non-obvious ways.

We need to move the query-condition logic over from TileDB-R before we can continue here.

**Comment by johnkerl** [comment]
This is a duplicate of #3061
